studio/lib/common/fetch/base.ts

```diff
@@ -86,15 +87,32 @@ export async function handleResponseError<T = unknown>(
   }
 }
 
+let currentSession: Session | null = null
+
+auth.onAuthStateChange((event, session) => {
+  currentSession = session
+})
```

Comment:
Is this 100% safe to call on the server? Could always do something like:
```suggestion
if (typeof window !== 'undefined') {
  auth.onAuthStateChange((event, session) => {
    currentSession = session
  })
}
```

Reply:
It is, it just registers a callback.